app/src/main/java/com/nutomic/syncthingandroid/views/SttracePreference.java

```diff
@@ -0,0 +1,100 @@
+package com.nutomic.syncthingandroid.views;
+
+import android.Manifest;
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+import android.preference.MultiSelectListPreference;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.AttributeSet;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * SttracePreference which allows the user to select which debug facilities
+ * are enabled.
+ *
+ * Setting can be "no debug facility" (none selected), or selecting individual  debug facilities.
+ *
+ * The preference is stored as Set&lt;String&gt; where an empty set represents
+ * "no debug facility".
+ *
+ * Debug facilities are documented in https://docs.syncthing.net/dev/debugging.html
+ *
+ */
+public class SttracePreference extends MultiSelectListPreference {
+
+    public SttracePreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        setDefaultValue(new TreeSet<String>());
+    }
+
+    public SttracePreference(Context context) {
+        this(context, null);
+    }
+
+    /**
+     * Show the dialog.
+     */
+    @Override
+    protected void showDialog(Bundle state) {
+        Set<String> selected = getSharedPreferences().getStringSet(getKey(), new HashSet<>());
+        // from JavaDoc: Note that you must not modify the set instance returned by this call.
+        // therefore required to make a defensive copy of the elements
+        selected = new HashSet<>(selected);
+        CharSequence[] all = getDebugFacilities();
+        filterRemovedDebugFacilities(selected, all);
+        setEntries(all);        // display without surrounding quotes
+        setEntryValues(all);    // the value of the entry is the debug facility "as is"
+        setValues(selected);    // the currently selected values
+        super.showDialog(state);
+    }
+
+    /**
+     * Removes any debug facility that is no longer present in the current syncthing version.
+     * Otherwise it will never be removed from the enabled facilities set by MultiSelectListPreference.
+     */
+    private void filterRemovedDebugFacilities(Set<String> selected, CharSequence[] all) {
+        HashSet<CharSequence> availableDebugFacilities = new HashSet<>(Arrays.asList(all));
+        selected.retainAll(availableDebugFacilities);
+    }
+
+    /**
+     * Returns all debug facilities available in the currently syncthing version.
+     */
+    private CharSequence[] getDebugFacilities() {
```

Comment:
I know, but syncthing is not running to get it so I'd go with the static. Its less effort submitting a PR for a new syncthing version that still has to be reviewed for changes than implementing a "caching from last successful run" thing. The API documentation is also outdated referring to st v12.
For me, as nobody cared about improving the debug options UI for years, the cut is here as I'm not willing to invest much more time for an asynchronous cached polling implementation. But if you have an easy-to-implement suggestion, I can help with that instead of making PRs with every new syncthing version that changes debug facilities.